File: 백준/Bronze/10101. 삼각형 외우기/삼각형 외우기.cc
#include<iostream>
#include<string>
#include<algorithm>
#include<queue>
#include <map>

using namespace std;

int main() {

	ios::sync_with_stdio(false);
	cin.tie(NULL);
    
	int n, m, k, result=0;
	
	cin >> n >> m >> k;

	result += n; result += m; result += k;

	if (n == 60 && m == 60 && k == 60)
		cout << "Equilateral";
	else if (result == 180 && n == m || n == k || m == k)
		cout << "Isosceles";
	else if (result == 180)
		cout << "Scalene";
	else
		cout << "Error";
}

File: 백준/Bronze/10101. 삼각형 외우기/삼각형 외우기.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		
		HashMap<Integer, Integer> angle = new HashMap<Integer, Integer>();
		
		Scanner sc = new Scanner(System.in);
		int result = 0;
		
		for (int i = 0; i < 3; i++) {
			int tmp = sc.nextInt();
			angle.put(tmp, 1);
			result += tmp;
		}
		
		
		if(result != 180) {
			System.out.print("Error");
		}
		else if(angle.size()==3) {
			System.out.print("Scalene");
		}
		else if(angle.size()==2) {
			System.out.print("Isosceles");
		}
		else if(angle.size()==1) {
			System.out.print("Equilateral");
		}
		
		

		
	}

}
